integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java

```diff
@@ -168,4 +170,18 @@ public void testJWTVerification() throws Exception {
         JWSVerifier verifier = new RSASSAVerifier(jwk.toRSAPublicKey());
         Assert.assertTrue(jwsObject.verify(verifier),"JWT failed to validate with JWKS response");
     }
+
+    @Test(description = "Test Rate limiting on JWKS", dependsOnMethods = "testJWTVerification")
+    public void testJWKSEndpointRatelimit() throws MalformedURLException, CCTestException {
+        boolean received429 = false;
+        for (int i = 0; i < 10; i++) {
+            HttpResponse res = HttpsClientRequest.doGet(Utils.getServiceURLHttps("/.wellknown/jwks"),
+                    new HashMap<>());
+            if (res.getResponseCode() == 429) {
```

Comment:
Don't we need to check that, `i >= 5`

Reply:
i=1 is already tested (from dependsOn method)